include/extractor/guidance/segregated_intersection_classification.hpp

```diff
@@ -0,0 +1,27 @@
+#include "util/typedefs.hpp"
+
+#include <unordered_set>
+
+namespace osrm
+{
+namespace util
+{
+class NameTable;
+}
+
+namespace extractor
+{
+class NodeBasedGraphFactory;
+
+namespace guidance
+{
+// Find all "segregated" edges, e.g. edges that can be skipped in turn instructions.
+// The main cases are:
+// - middle edges between two osm ways in one logic road (U-turn)
+// - staggered intersections (X-cross)
+// - square/circle intersections
+std::unordered_set<EdgeID> findSegregatedNodes(NodeBasedGraphFactory &factory, const util::NameTable& names);
```

Comment:
Should factory be const? If not why not? Not clear from interface.

Reply:
Good catch, turns out the interface of `NodeBasedGraphFactory` did not apply const correctly.